pkg/operator/controller/dns/controller.go

```diff
@@ -141,22 +152,22 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				LastTransitionTime: metav1.Now(),
 			})
 			result.RequeueAfter = 30 * time.Second
+		} else {
+			conditions = append(conditions, iov1.DNSZoneCondition{
+				Type:               iov1.DNSRecordFailedConditionType,
+				Status:             string(operatorv1.ConditionFalse),
+				Reason:             "ProviderSuccess",
+				Message:            fmt.Sprintf("The DNS provider succeeded to ensure the record: %v", err),
+				LastTransitionTime: metav1.Now(),
+			})
 		}
+
 		statuses = append(statuses, iov1.DNSZoneStatus{
 			DNSZone:    zone,
 			Conditions: conditions,
 		})
 	}
-
-	// TODO: only update if status changed
-	updated := record.DeepCopy()
-	updated.Status.Zones = statuses
-	if err := r.client.Status().Update(context.TODO(), updated); err != nil {
-		log.Error(err, "failed to update dnsrecord; will retry", "dnsrecord", record)
-		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
-	}
-
-	return result, nil
+	return statuses, result
```

Comment:
What do you think about using `retryableerror.New` in case publishing to one or more of the zones failed?